i18n/zh-CN/docusaurus-plugin-content-blog/2024-03-04-kcl-0.8.0-release/index.md

```diff
@@ -0,0 +1,443 @@
+
+```shell
+Hello World !
+```
+
+编译结果
+
+```shell
+a: Hello World !
+```
+
+通过结合 `json.decode` 方法，可以轻松的实现将 json 文件反序列化。
+
+在 `hello.json` 文件中增加如下内容
+
+```json
+{
+  "name": "John",
+  "age": 10
+}
+```
+
+KCL 程序如下所示：
+
+```kcl
+import file
+import json
+
+_a = json.decode(file.read("hello.json"))
+
+name = _a.name
+age = _a.age
+```
+
+编译结果如下所示：
+
+```shell
+name: John
+age: 10
+```
+
+更多内容 - [https://kcl-lang.io/zh-CN/docs/reference/model/file/](https://kcl-lang.io/zh-CN/docs/reference/model/file/)
+
+#### KCL 编译缓存路径支持使用环境变量 KCL_CACHE_PATH 指定
+
+KCL 编译器会将缓存生成到环境变量`KCL_CACHE_PATH`指定的目录当中，如果没指定，将会生成到项目根目录中。
+
+#### 插件系统支持使用 golang 编写 KCL 插件
+
+使用 golang 定义 hello 插件。
+
+```golang
+package hello_plugin
+
+import (
+    "kcl-lang.io/kcl-go/pkg/plugin"
+)
+
+func init() {
+    plugin.RegisterPlugin(plugin.Plugin{
+        Name: "hello",
+        MethodMap: map[string]plugin.MethodSpec{
+            "add": {
+                Body: func(args *plugin.MethodArgs) (*plugin.MethodResult, error) {
+                    v := args.IntArg(0) + args.IntArg(1)
+                    return &plugin.MethodResult{V: v}, nil
+                },
+            },
+        },
+    })
+}
+```
+
+借助 kcl-go 开发，扩展 KCL 编译器使用插件。
+
+```kcl
+package main
+
+import (
+    "fmt"
+
+    "kcl-lang.io/kcl-go/pkg/kcl"
+    "kcl-lang.io/kcl-go/pkg/native"                // Import the native API
+    _ "kcl-lang.io/kcl-go/pkg/plugin/hello_plugin" // Import the hello plugin
+)
+
+func main() {
+    // Note we use `native.MustRun` here instead of `kcl.MustRun`, because it needs the cgo feature.
+    yaml := native.MustRun("main.k", kcl.WithCode(code)).GetRawYamlResult()
+    fmt.Println(yaml)
+}
+
+const code = `
+import kcl_plugin.hello
+
+name = "kcl"
+three = hello.add(1,2) # 3
+```
+
+### 😸 新增 Linux arm64 版本
+
+KCL 的 Release 产物中新增了对 Linux arm64 平台的支持。
+
+可以在 [KCL Release Page](https://github.com/kcl-lang/kcl/releases) 中找到后缀为 `linux-arm64` 的压缩包。
+
+### 🏄 SDK & API 更新
+
+#### Rust SDK 
+
+KCL Rust SDK 提供了一系列的 API，可以用于 KCL 文件的编译、校验、测试和格式化代码等操作。
+
+[KCL Rust SDK: https://github.com/kcl-lang/lib](https://github.com/kcl-lang/lib)
+
+#### Java SDK
+
+KCL Java SDK 新增语法树、作用域、符号等语法语义结构定义及相关查询 API。
+
+#### Go SDL 更新
+
+- KCL Doc 工具支持输出为 OpenAPI 格式。
+- 增加 Parse 过程相关 API。
+
+#### API 更新
+
+- 新增语法和语义分析 API 用于对 KCL 代码进行分析。
+- 新增构建二进制产物 API 用于缓存编译结果。
+- 新增运行二进制产物 API 用于直接运行编译结果，避免重复编译并提升性能。
+- 新增代码生成 API 以编程方式实现 KCL 代码生成而不是编写复杂的模版。
+
+### 🐞 其他更新及错误修复
+
+- 修复了 KCL CLI 使用编译参数 -S 可能会导致的编译错误
+- 修复了 kcl fmt 工具对 lambda 表达式进行格式化时，结尾增加一个空行的错误。
+- 修复 Schema Doc 补全代码片段错误
+- 优化非预期 token 的语法错误提示
+- 去除 Schema 对象内部非预期的内置类型属性通过 print 输出
+- 修复非预期的字典生成表达式中的 key 与循环变量相同时的变量计算
+- 修复 schema 内部诸如 "$if" 的字符串标识符定义找不到的错误
+
+## IDE & 工具链更新
+
+### IDE 更新
+
+#### IDE 语义级别的高亮增强
+
+KCL IDE 高亮之前仅支持 KCL 语法高亮，如下图所示：
+
+![](/img/blog/2024-01-18-biweekly-newsletter/old-ide.png)
+
+我们今年逐步启用了新的 KCL 语义架构模型，在新语义架构的支撑下，KCL IDE 支持语义级别的高亮，在语义上有关联的代码会显示相同的高亮。
+
+![](/img/blog/2024-01-18-biweekly-newsletter/new-ide.png)
```

Comment:
Supplementing complete IDE changes including kcl-lsp VS Code and idea.

Reply:
fixed